llvm/lib/Target/AMDGPU/SIInstructions.td

```diff
@@ -3553,6 +3553,23 @@ def : AMDGPUPat <
   (V_BFE_U32_e64 $src, (i32 0), $width)
 >;
 
+def uint5Bits : PatLeaf<(i32 VGPR_32:$width), [{
+  return CurDAG->computeKnownBits(SDValue(N, 0)).countMaxTrailingOnes() <= 5;
```

Comment:
No, this is completely wrong!!! E.g. if bit 0 is known to be 0 then `countMaxTrailingOnes` will return 0 and this test will pass.

Reply:
Ugh, my bad, sorry.

Reply:
Thanks so much.  Also countMaxActiveBits is more close to what we are going to define